extensions/amp-sidebar/1.0/sidebar.js

```diff
@@ -89,6 +89,7 @@ function SidebarWithRef(
   const classes = useStyles();
   const sidebarRef = useRef();
   const maskRef = useRef();
+  const sideRef = useRef(null);
```

Comment:
In this case I'm curious if `useRef` is preferred to `useState` since IMO it would be reasonable to rerender in a new location entirely when the `side` prop or document direction changes.

Reply:
In my opinion, I believe `useRef` makes sense in the current design.  Essentially we need to get a ref for the element that is rendered (`sidebarRef`) in order to access the `document` and we only have access to this within `useLayoutEffect`.

Once we are within `useLayoutEffect`, the animation immediately runs and also needs to know whether it should animate from the left or the right side, so at this point all the calculations for side are complete and stored within `sideRef` and the animation immediately runs.  At this point, we do not want to trigger a re-render since the animation and everything is done.

With the case where `side` prop is changed, the component actually does re-render due to the prop change so this is covered.

With the case of the document direction changes, this would not be captured, but seems like too obscure of a use case to design for, especially since we only rely on the document direction if the user does not provide the `side` prop.

Reply:
I actually think that `useState` is a more correct form with the `className`  computation. Otherwise, the new classname can be overwritten at any time. Also it's a logic construct and other things might depend on it eventually.

Reply:
To clarify what I was thinking of, one way we can compute `className` declaratively by having a state for document direction separate from `side` (and this could be another use case for a built in `useDir` or similar).
```js
function Sidebar({side}) {
  const [rtl, setRtl] = useState(false);
  side = isEnumValue(side) && side ?? rtl ? Side.RIGHT : Side.LEFT;

  useLayoutEffect(() => {
    const sidebarElement = sidebarRef.current;
    if (!sidebarElement) {
      return;
    }
    const doc = sidebarElement.ownerDocument;
    if (!doc) {
      return;
    }
    setRtl(isRTL(doc));
  });

  return (
    <ContainWrapper 
      className={`${classes.sidebarClass} 
        ${side === Side.LEFT ? classes.left : classes.right}`} />
  );
}
```

There are other ways, as well. We could do: 
```js
/** @private @enum {string} */
const Side = {
  LEFT: 'left',
  RIGHT: 'right',
  AUTO: 'auto'
};

function Sidebar({side = Side.AUTO}) {
  return (
    <ContainWrapper 
      className={`${classes.sidebarClass} ${classes[side]}`} ${classes[dir]}/>
  );
}
```

With the following JSS:
```js
const left = {
  left: 0,
};

const right = {
  right: 0,
};

const ltr = {}
const rtl = {}
const auto = {
  '&$ltr': {
    left: 0,
  },
  '&$rtl': {
    right: 0
  }
}
```

I recall a prior issue against using bracket notation on classes but unfortunately don't remember exactly why we didn't end up doing it.